### PR TITLE
[NEW] Support sending audio records

### PR DESCRIPTION
**Pull request by luismachado**
@RocketChat/ios

I forked #762 (as no update was given since a month ago)  so credit also to @falcaoaugustos (maybe it's best to update his PR with this one?)

Changes since the fork:
- Visual feedback is different now (whole screen is blurred while recording and a timer is also present)
- Feedback is now quicker as I replaced the button action for a tapgesturerecognizer (`delayTouchesBegan` was also removed)
- Code is also up to date with the most recent commits.

### CODE REVIEW

**Comment by rafaelks** [comment]
@luismachado Looks like this one have conflicts? 🤔

**Comment by luismachado** [comment]
@rafaelks Yeah also noticed that. Looking into it now.

**Comment by luismachado** [comment]
@rafaelks should be without conflicts now. Let me know what you think :)

File: Rocket.Chat.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -1691,7 +1700,7 @@
 				TargetAttributes = {
 					41DF76DE1D2C50710028DBF8 = {
 						CreatedOnToolsVersion = 7.3.1;
-						DevelopmentTeam = S6UPZG7ZR3;
+						DevelopmentTeam = MJHBLUH3VK;
```
Comment (rafaelks) [inline Rocket.Chat.xcodeproj/project.pbxproj]: Can you remove this change?

File: Rocket.Chat.xcodeproj/project.pbxproj [hunk 2]
```diff
@@ -2543,12 +2555,12 @@
 				CODE_SIGN_ENTITLEMENTS = Rocket.Chat/Rocket.Chat.entitlements;
 				CODE_SIGN_IDENTITY = "iPhone Developer";
 				"CODE_SIGN_IDENTITY[sdk=iphoneos*]" = "iPhone Developer";
-				DEVELOPMENT_TEAM = S6UPZG7ZR3;
+				DEVELOPMENT_TEAM = MJHBLUH3VK;
```
Comment (rafaelks) [inline Rocket.Chat.xcodeproj/project.pbxproj]: Can you remove this change?

File: Rocket.Chat.xcodeproj/project.pbxproj [hunk 3]
```diff
@@ -2543,12 +2555,12 @@
...
 				INFOPLIST_FILE = Rocket.Chat/Info.plist;
 				IPHONEOS_DEPLOYMENT_TARGET = 10.0;
 				LD_RUNPATH_SEARCH_PATHS = "$(inherited) @executable_path/Frameworks";
 				OTHER_SWIFT_FLAGS = "$(inherited) -D DEBUG \"-D\" \"COCOAPODS\"";
-				PRODUCT_BUNDLE_IDENTIFIER = chat.rocket.ios;
+				PRODUCT_BUNDLE_IDENTIFIER = chat.rocket.ios11;
```
Comment (rafaelks) [inline Rocket.Chat.xcodeproj/project.pbxproj]: We also need to remove this change.

File: Rocket.Chat.xcodeproj/project.pbxproj [hunk 4]
```diff
@@ -2565,11 +2577,11 @@
 				CODE_SIGN_ENTITLEMENTS = Rocket.Chat/Rocket.Chat.entitlements;
 				CODE_SIGN_IDENTITY = "iPhone Developer";
 				"CODE_SIGN_IDENTITY[sdk=iphoneos*]" = "iPhone Developer";
-				DEVELOPMENT_TEAM = S6UPZG7ZR3;
+				DEVELOPMENT_TEAM = MJHBLUH3VK;
```
Comment (rafaelks) [inline Rocket.Chat.xcodeproj/project.pbxproj]: Can you remove this change?

File: Rocket.Chat.xcodeproj/project.pbxproj [hunk 5]
```diff
@@ -2565,11 +2577,11 @@
...
 				INFOPLIST_FILE = Rocket.Chat/Info.plist;
 				IPHONEOS_DEPLOYMENT_TARGET = 10.0;
 				LD_RUNPATH_SEARCH_PATHS = "$(inherited) @executable_path/Frameworks";
-				PRODUCT_BUNDLE_IDENTIFIER = chat.rocket.ios;
+				PRODUCT_BUNDLE_IDENTIFIER = chat.rocket.ios11;
```
Comment (rafaelks) [inline Rocket.Chat.xcodeproj/project.pbxproj]: Can you remove this change?

File: Rocket.Chat/en.lproj/Localizable.strings [hunk 6]
```diff
@@ -28,6 +28,13 @@
 "alert.settings.set_mail_app.title" = "Oops!";
 "alert.settings.set_mail_app.message" = "No Mail Accounts. Please set up a Mail account in order to send email.";
 
+"alert.audio_message.success.send" = "Send";
+"alert.audio_message.success.title" = "Success";
+"alert.audio_message.success.message" = "You message was successfully recorded. Recording time: %@.";
+
+"alert.audio_message.error.title" = "Ooops!";
+"alert.audio_message.error.message" = "An error has occurred. Please, record your message again.";
+
```
Comment (rafaelks) [inline Rocket.Chat/en.lproj/Localizable.strings]: We've to add this strings to the other localizable files (in English is fine if you don't know the other languages). Is just to keep it in sync.

**Comment by luismachado** [comment]
@rafaelks Okay, just addressed your remarks. Regarding the translation, I was going to add the PT version but it looks like it was already there :P 

Therefore I just added the lines for the other 3 but left a TODO tag near each one, as I don't trust google translate with longer phrases...

**Comment by rafaelks** [comment]
Looking good, but I noticed something curious... always, the first time I send the audio, it doesn't work, stays with 00:00 after the upload. Can you also reproduce this issue?

**Comment by luismachado** [comment]
Yeah, also just noticed that. It's also happening in @falcaoaugustos branch. 
It looks to always be the first recording per launch (even if you don't send the 1st one the 2nd one will be okay).

Going to try to find out what's happening.

**Comment by luismachado** [comment]
Okay, the issue seems now seems to be fixed. 

Would you care to check it on your end @rafaelks ? Thanks.

**Comment by rafaelks** [comment]
Thank you @luismachado & @falcaoaugustos!!! 🚀

**Comment by falcaoaugustos** [comment]
Thank you very much @luismachado . I had a few problems fixing it but I will read your code in order to learn how to do so. I must say that I am still studying this case.
Thank you too, @rafaelks . By the way, I will make more contributions as soon as I solve my money issue.

**Pull request merged**